discovery-provider/src/solana/audius_data_transaction_handlers.py

```diff
@@ -0,0 +1,394 @@
+
+
+class InitAdminData(TypedDict):
+    authority: Pubkey
+    verifier: Pubkey
+
+
+def handle_init_admin(
+    session: Session,
+    transaction: ParsedTx,
+    instruction: ParsedTxInstr,
+    db_models: Dict,
+    metadata_dictionary: Dict,
+    records: List[Any],
+):
+    pass
+
+
+class UpdateAdminData(TypedDict):
+    is_write_enabled: bool
+
+
+def handle_update_admin(
+    session: Session,
+    transaction: ParsedTx,
+    instruction: ParsedTxInstr,
+    db_models: Dict,
+    metadata_dictionary: Dict,
+    records: List[Any],
+):
+    pass
+
+
+class InitUserData(TypedDict):
+    base: Pubkey
+    eth_address: List[int]
+    replica_set: List[int]
+    _replica_set_bumps: List[int]
+    handle_seed: int
+    _user_bump: int
+    _metadata: str
+
+
+def handle_init_user(
+    session: Session,
+    transaction: ParsedTx,
+    instruction: ParsedTxInstr,
+    db_models: Dict,
+    metadata_dictionary: Dict,
+    records: List[Any],
+):
+    # NOTE: Not sure if we should update the user's model until they claim account
+    # user_id = instruction.get("data").get("id")
+    # existing_user = db_models.get("users", {}).get(user_id)
+    # user = User(**existing_user.asdict())
+    # user.slot = transaction["result"]["slot"]
+
+    # replica_set = list(instruction.get("data").get("replica_set"))
+    # user.primary_id = replica_set[0]
+    # user.secondary_ids = [replica_set[1], replica_set[2]]
+
+    # metadata_cid = instruction.get('data').get('metadata')
+
+    # No action to be taken here
+    pass
+
+
+def handle_init_user_sol(
+    session: Session,
+    transaction: ParsedTx,
+    instruction: ParsedTxInstr,
+    db_models: Dict,
+    metadata_dictionary: Dict,
+    records: List[Any],
+):
+    pass
+
+
+class CreateUserData(TypedDict):
+    base: Pubkey
+    eth_address: List[int]
+    replica_set: List[int]
+    replica_set_bumps: List[int]
+    user_id: int
+    user_bump: int
+    metadata: str
+    user_authority: Pubkey
+
+
+def handle_create_user(
+    session: Session,
+    transaction: ParsedTx,
+    instruction: ParsedTxInstr,
+    db_models: Dict,
+    metadata_dictionary: Dict,
+    records: List[Any],
+):
+    instruction_data: CreateUserData = instruction["data"]
+    # Validate that the user row doesn't already exist - error if it does
+    user_id = instruction_data["user_id"]
+    # TODO: validate uniqueness on handle
+    replica_set = list(instruction_data["replica_set"])
+
+    eth_address = Web3.toChecksumAddress(
+        f"0x{bytes(list(instruction_data['eth_address'])).hex()}"
+    )
+    metadata_multihash = instruction_data.get("metadata")
+
+    user = User(
+        blockhash=None,
+        blocknumber=None,
+        slot=transaction["result"]["slot"],
+        user_storage_account=str(instruction.get("account_names_map").get("user")),
+        user_authority_account=str(instruction_data.get("user_authority")),
+        txhash=transaction["tx_sig"],
+        user_id=user_id,
+        is_current=True,
```

Comment:
should we add creator_node_endpoint here?

Reply:
Thoughts on pulling it from the metadata instead of converting spid into urls?